### PR TITLE
GUI config rules

### PR DESCRIPTION
**Pull request by HEdingfield**
More progress on #45. Reviewing one commit at a time might be easiest.

### CODE REVIEW

File: src/main/java/com/rcv/GuiConfigController.java [hunk 1]
```diff
@@ -192,10 +239,24 @@ private void saveElectionConfig(File saveFile) {
...
+            : Tabulator.TieBreakMode.MODE_UNKNOWN.toString();
     rules.overvoteRule =
         choiceOvervoteRule.getValue() != null
             ? choiceOvervoteRule.getValue().toString()
             : Tabulator.OvervoteRule.RULE_UNKNOWN.toString();
+    rules.multiSeatTransferRule =
+        choiceMultiSeatTransferRule.getValue() != null
+            ? choiceMultiSeatTransferRule.getValue().toString()
+            : Tabulator.MultiSeatTransferRule.TRANSFER_RULE_UNKNOWN.toString();
+    rules.maxRankingsAllowed = Integer.parseInt(textFieldMaxRankingsAllowed.getText());
```
Comment (tarheel) [inline src/main/java/com/rcv/GuiConfigController.java]: Will these trigger an uncaught exception if the text isn't a valid integer? Do your listeners make that truly impossible, or just unlikely?

File: src/main/java/com/rcv/GuiConfigController.java [hunk 2]
```diff
@@ -192,36 +200,56 @@ public LocalDate fromString(String string) {
...
-    textFieldMinimumVoteThreshold.textProperty().addListener((observable, oldValue, newValue) -> {
-      if (!newValue.matches("\\d*")) {
-        textFieldMinimumVoteThreshold.setText(oldValue);
-      }
-    });
+    // Restrict text fields to non-negative integers and set default values where applicable
+    textFieldMaxRankingsAllowed
+        .textProperty()
+        .addListener(
+            (observable, oldValue, newValue) -> {
```
Comment (tarheel) [inline src/main/java/com/rcv/GuiConfigController.java]: If you can access the field itself from the listener args (maybe via observable?), perhaps you can avoid duplicating this same logic for each field? The amount of repetition here is unfortunate.

File: src/main/java/com/rcv/GuiConfigController.java [hunk 3]
```diff
@@ -251,12 +279,40 @@ private void saveElectionConfig(File saveFile) {
...
+    rules.decimalPlacesForVoteArithmetic =
+        !textFieldDecimalPlacesForVoteArithmetic.getText().isEmpty()
+            ? Integer.parseInt(textFieldDecimalPlacesForVoteArithmetic.getText())
+            : ElectionConfig.DEFAULT_DECIMAL_PLACES_FOR_VOTE_ARITHMETIC;
+    rules.minimumVoteThreshold =
+        !textFieldMinimumVoteThreshold.getText().isEmpty()
+            ? Integer.parseInt(textFieldMinimumVoteThreshold.getText())
+            : ElectionConfig.DEFAULT_MINIMUM_VOTE_THRESHOLD.intValue();
+    rules.batchElimination =
+        ((RadioButton) toggleBatchElimination.getSelectedToggle()).getText().equals("True");
```
Comment (tarheel) [inline src/main/java/com/rcv/GuiConfigController.java]: Could also factor out this bit of repeated logic.

File: src/main/java/com/rcv/GuiConfigController.java (hunk 1 above)
Comment (HEdingfield) [reply src/main/java/com/rcv/GuiConfigController.java]: Yes, it's impossible. Anything that doesn't match regex "\\d*" in these text fields is immediately reverted, as if it was never typed at all. Maybe not the best UX (ideally we'd have a pop-up note that says "integers only"), but it'll do the trick in the short-term.

File: src/main/java/com/rcv/GuiConfigController.java (hunk 2 above)
Comment (HEdingfield) [reply src/main/java/com/rcv/GuiConfigController.java]: Wasn't possible via observable unfortunately, but I made a new inner listener class in the latest commit.

File: src/main/java/com/rcv/GuiConfigController.java (hunk 3 above)
Comment (HEdingfield) [reply src/main/java/com/rcv/GuiConfigController.java]: Done in latest commit, along with additional refactoring. Please re-approve when you get a chance to review. 💯

**Pull request merged**